packages/front-end/components/HealthTab/DimensionIssues.tsx

```diff
@@ -0,0 +1,264 @@
+import { ExperimentSnapshotTrafficDimension } from "back-end/types/experiment-snapshot";
+import { ExperimentReportVariation } from "back-end/types/report";
+import { useState } from "react";
+import { useUser } from "@/services/UserContext";
+import { DEFAULT_SRM_THRESHOLD } from "@/pages/settings";
+import track from "@/services/track";
+import { pValueFormatter } from "@/services/experiments";
+import VariationUsersTable from "../Experiment/TabbedPage/VariationUsersTable";
+import Modal from "../Modal";
+import SelectField from "../Forms/SelectField";
+import { EXPERIMENT_DIMENSION_PREFIX, srmHealthCheck } from "./SRMDrawer";
+import HealthCard from "./HealthCard";
+import { IssueTags } from "./IssueTags";
+
+interface Props {
+  dimensionData: {
+    [dimension: string]: ExperimentSnapshotTrafficDimension[];
+  };
+  variations: ExperimentReportVariation[];
+}
+
+type NewObjectArray = {
+  value: string;
+  label: string;
+  issues: string[];
+};
+
+export function transformDimensionData(
+  dimensionData: {
+    [dimension: string]: ExperimentSnapshotTrafficDimension[];
+  },
+  variations: ExperimentReportVariation[],
+  srmThreshold: number
+): NewObjectArray[] {
+  return Object.entries(dimensionData).flatMap(
+    ([dimensionName, dimensionSlices]) => {
+      // Skip for date dimension
+      if (dimensionName === "dim_exposure_date") {
+        return [];
+      }
+
+      const dimensionSlicesWithIssues = dimensionSlices.filter((item) => {
+        const totalDimUsers = item.variationUnits.reduce(
+          (acc, a) => acc + a,
+          0
+        );
+        return (
+          srmHealthCheck({
+            srm: item.srm,
+            variations,
+            srmThreshold,
+            totalUsers: totalDimUsers,
+          }) !== "healthy"
+        );
+      });
+
+      const issueNames = dimensionSlicesWithIssues.map((item) => item.name);
+
+      // Construct the new object for the current dimension
+      return {
+        value: dimensionName,
+        label: dimensionName.replace(EXPERIMENT_DIMENSION_PREFIX, ""),
+        issues: issueNames,
+      };
+    }
+  );
+}
+
+export const DimensionIssues = ({ dimensionData, variations }: Props) => {
+  const { settings } = useUser();
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const srmThreshold = settings.srmThreshold ?? DEFAULT_SRM_THRESHOLD;
+
+  const availableDimensions = transformDimensionData(
+    dimensionData,
+    variations,
+    srmThreshold
+  );
+
+  const [selectedDimension, setSelectedDimension] = useState(
+    availableDimensions[0].value
```

Comment:
I got an error here, I think sometimes this can be empty.

I think we probably need to handle this at a higher level than this component, but this component should also be robust to this array having no length